run/gem5art/run.py

```diff
@@ -0,0 +1,516 @@
+        """
+        run = cls()
+        run.gem5_binary = gem5_binary
+        run.run_script = run_script
+        run.gem5_artifact = gem5_artifact
+        run.gem5_git_artifact = gem5_git_artifact
+        run.run_script_git_artifact = run_script_git_artifact
+        run.params = params
+        run.timeout = timeout
+
+        run._id = uuid4()
+
+        # Assumes **/<gem5_name>/gem5.<anything>
+        run.gem5_name = os.path.split(os.path.split(run.gem5_binary)[0])[1]
+        # Assumes **/<script_name>.py
+        run.script_name = os.path.split(run.run_script)[1].split('.')[0]
+
+        # Info about the actual run
+        run.running = False
+        run.enqueue_time = time.time()
+        run.start_time = 0.0
+        run.end_time = 0.0
+        run.return_code = 0
+        run.kill_reason = ''
+        run.status = "Created"
+        run.pid = 0
+        run.task_id = None
+
+        # Initially, there are no results
+        run.results = None
+
+        return run
+
+    @classmethod
+    def createSERun(cls,
+                    gem5_binary: str,
+                    run_script: str,
+                    gem5_artifact: Artifact,
+                    gem5_git_artifact: Artifact,
+                    run_script_git_artifact: Artifact,
+                    *params: str,
+                    timeout: int = 60*15) -> 'gem5Run':
+        """
+        gem5_binary and run_script are the paths to the binary to run
+        and the script to pass to gem5. Full paths are better.
+
+        The artifact parameters (gem5_artifact, gem5_git_artifact, and
+        run_script_git_artifact) are used to ensure this is reproducible run.
+
+        Further parameters can be passed via extra arguments. These
+        parameters will be passed in order to the gem5 run script.
+        timeout is the time in seconds to run the subprocess before killing it.
+
+        Note: When instantiating this class for the first time, it will create
+        a file `info.json` in the outdir which contains a serialized version
+        of this class.
+        """
+
+        run = cls._create(gem5_binary, run_script, gem5_artifact,
+                          gem5_git_artifact, run_script_git_artifact, params,
+                          timeout)
+
+        run.artifacts = [gem5_artifact, gem5_git_artifact,
+                         run_script_git_artifact]
+
+        run.string = f"{run.gem5_name} {run.script_name}"
+        run.string += ' '.join(run.params)
+
+        # Constructs the output directory from the gem5 build opts (e.g., X86,
+        # ARM, MOESI_hammer), the gem5 run script, and the extra parameters.
+        # The relative path is useful for printing
+        run.relative_outdir = os.path.join('results',
+                                           run.gem5_name,
+                                           run.script_name,
+                                           *run.params)
+
+        run.outdir = os.path.abspath(run.outdir)
```

Comment:
I think, this should be run.outdir = os.path.abspath(run.relative_outdir)

Reply:
Yep. Fixed in #27!